packages/cli/src/evaluation.ee/test-runner/test-runner.service.ee.ts

```diff
@@ -401,6 +414,24 @@ export class TestRunnerService {
 		return triggerOutput;
 	}
 
+	private getEvaluationData(
+		execution: IRun,
+		workflow: IWorkflowBase,
+		operation: 'setInputs' | 'setOutputs',
+	): IDataObject {
+		const evalNodes = TestRunnerService.getEvaluationNodes(workflow, operation);
+
+		return evalNodes.reduce((accu: IDataObject, node: INode) => {
+			const runs = execution.data.resultData.runData[node.name];
+			const data = runs?.[0]?.data?.[NodeConnectionTypes.Main]?.[0]?.[0]?.evaluationData ?? {};
+
+			return {
+				...accu,
+				...data,
+			};
+		}, {} as IDataObject);
```

Comment:
1. `reduce` takes a [generic](https://share.cleanshot.com/hs6RTy69) to type the accumulator, so no need to type it in the first arg to the callback and in the second arg.
2. The second arg to the `reduce` callback is inferrable from `evalNodes`.
3. Can we avoid using `IDataObject` please?
4. How much data is `evaluationData`? If not little, I typically avoid using `reduce` with spreads like this as it can generate a lot of garbage to collect, as opposed to mutating the accumulator.

Reply:
1. Will update once we agree on a type. That's good feedback.
4. However much data the user deems important to record for evalutations.